File: server/Optika.API/Optika.API/Entities/Brand.cs
using System.ComponentModel.DataAnnotations;

namespace Optika.API.Entities
{
    public class Brand
    {
        public int Id { get; set; }

        [Required, MaxLength(100)]
        public string Name { get; set; } = null!;

        [MaxLength(100)]
        public string? Country { get; set; }

        public ICollection<Product> Products { get; set; } = new List<Product>();

    }
}

File: client/lib/models/brand.dart
class Brand {
  final int id;
  final String name;
  final String? country;

  Brand({
    required this.id,
    required this.name,
    this.country,
  });

  // Фабричный метод для создания объекта из JSON
  factory Brand.fromJson(Map<String, dynamic> json) {
    return Brand(
      id: json['id'],
      name: json['name'],
      country: json['country'],
    );
  }

  // Метод для преобразования объекта в JSON
  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'country': country,
    };
  }
}
